components/org.wso2.carbon.identity.data.publisher.authentication.analytics.login/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/analytics/login/AnalyticsLoginDataPublisherUtils.java

```diff
@@ -185,8 +189,16 @@ public static AuthenticationData buildAuthnDataForAuthentication(Event event) {
         authenticationData.setEventType(AnalyticsLoginDataPublishConstants.OVERALL_EVENT);
         authenticationData.setContextId(context.getContextIdentifier());
         authenticationData.setEventId(UUID.randomUUID().toString());
-        authenticationData.setAuthnSuccess(true);
-        authenticationData.setRemoteIp(IdentityUtil.getClientIpAddress(request));
+        if (status.equals(AuthenticatorStatus.PASS)) {
```

Comment:
Can status be null? Lets change the equals method to `AuthenticatorStatus.PASS.equals(status)` to avoid any accidental null pointers.

Reply:
Modified.